mvp-spec/audit-log-design.md

```diff
@@ -0,0 +1,257 @@
+| List of the Preferences          | Each Preference must be paired with the name of the CMP who generated it and signed it with a Red/Green indicator expressing the validity of the signature.                                                             |
+| Seed                             | The Seed is represented by the Transaction ID and a Red/Green indicator expressing the validity of the signature                                                                                                             |
+| List of Transmission Results     | The Transmission Results available in the Audit Log. Each Transmission Results is represented by the Name of the Receiver, the status of the Transmission and a Red/Green indicator expressing the validity of the signature |
+
+The design of the page is up to the Contracting Party but it must consider 
+to have a neutral UI to be used for many different Publishers in case of a DSP.
+
+For the version v0.1, there are only one possible preference
+("opt-in") and one possible pseudonymous-identifiers ("prebid_id").
+However the framework is designed to handle many of them. 
+
+## Source validations
+
+As we saw, for displaying correctly the Audit UI, the Contracting Party must 
+validate different sources. For each source available in the 
+Audit Log, the Contracting Party must:
+
+* Call the Identity Endpoint of the "domain" of the source to retrieve its name
+  and its public key used for verifying the signature.
+* Decode the signature thanks to the NIST P-256 public key of the domain that
+  generated it.
+* Build a UTF-8 string based on the data available in the Audit Log. For each
+  data that is signed, there is a specific way to generate this UTF-8 string.
+  Those generations are described in this document.
+* Generate an SHA-256 hash from the UTF-8 string.
+* Compare the hash to the decoded signature. If they match, the signature is
+  validated.
+
+### Verify the Pseudo-Identifiers
+
+The Audit Log contains a list of Pseudo-identifiers. Each Pseudo-identifier is
+signed. The UTF-8 string for a specific Pseudo-Identifier must be built as follows:
+
+<!--partial-begin { "files": [ "identifier-signature-string.txt" ], "block": "" } -->
+<!-- ⚠️ GENERATED CONTENT - DO NOT MODIFY DIRECTLY ⚠️ -->
+```
+identifier.source.domain + '\u2063' + 
+identifier.source.timestamp + '\u2063' + 
+identifier.type + '\u2063'+
+identifier.value
+
+```
+<!--partial-end-->
+
+### Verify the Preferences
+
+The Audit Log contains a list of Preferences with one signature. The UTF-8 string for a specific Preference must be built as follows:
+
+<!--partial-begin { "files": [ "preferences-signature-string.txt" ], "block": "" } -->
+<!-- ⚠️ GENERATED CONTENT - DO NOT MODIFY DIRECTLY ⚠️ -->
+```
+preferences.source.domain + '\u2063' +
+preferences.source.timestamp + '\u2063' +
+prebid_id.source.signature + '\u2063' +
```

Comment:
Should add a comment to explain what "prebid_id" is (the identifier that has type "prebid_id")

Reply:
I tweak the signature description to make it more understable.

Reply:
Thanks, I like it!
However, it seems you've updated the main doc but not the partial (or at least, I like the "main" version better than the partial one): `identifiers[type=prebid_id]`